bchydro/api.py

```diff
@@ -54,6 +56,18 @@ async def authenticate(self, username, password) -> bool:
             try:
                 soup = BeautifulSoup(text, features="html.parser")
                 self._bchydroparam = soup.find(id="bchydroparam").text
+                # If the user has multiple accounts (eg. after a move), pick the first open one
+                # todo: allow user to specify
+                accountListDivs = soup.find_all("div", class_="accountListDiv")
+                if len(accountListDivs) > 0:
+                    accounts_response = await session.post(URL_GET_ACCOUNTS, headers={'x-csrf-token': self._bchydroparam})
+                    accounts = await accounts_response.json()
+                    account_id = accounts['accounts'][0]['accountId']
+                    response = await session.get(URL_ACCOUNTS_OVERVIEW + "?aid=" + account_id)
+                    text = await response.text()
+                    soup = BeautifulSoup(text, features="html.parser")
+                    self._bchydroparam = soup.find(id="bchydroparam").text
```

Comment:
I'd like to figure out if the `bchydroparam` is present more often as an `<input />` element or a `<span />` element. During testing I found that there were instances where it was present as both, and other instances where it was only present as an input.

If this param is always available as an input, we could use the following code to fetch the value:

```
self._bchydroparam = soup.find(attrs={"name": "bchydroparam"})['value']
```

If we need to test for both cases it might make sense to break this out to a new method.

Reply:
Sure, new method now tests both.